pkg/agents/baseagent.go

```diff
@@ -30,25 +28,32 @@ func NewBaseAgent(abm *abm.ABM, floor, hp int) (*Base, error) {
 		return nil, errors.New("Agent needs a Tower world to operate")
 	}
 	return &Base{
-		floor: floor,
-		hp:    hp,
 		tower: tower,
-		id:    uuid.New().String(),
+		id:    uuid,
 	}, nil
 }
 
 func (a *Base) Run() {
-	log.Printf("An agent cycle executed from base agent %d", a.floor)
+	floor := a.tower.GetFloor(a.id)
+	log.Printf("An agent cycle executed from base agent %d", floor)
 }
 
 func (a *Base) HP() int {
-	return a.hp
+	return a.tower.GetHP(a.id)
 }
 
 func (a *Base) Floor() int {
-	return a.floor
+	return a.tower.GetFloor(a.id)
 }
 
 func (a *Base) ID() string {
 	return a.id
 }
+
+func (a *Base) IsDead() bool {
```

Comment:
This is a big issue, overrides can basically be done implicitly on the original `IsDead` function and return wrong states. We should find a way to make these functions final or non-writeable

<img width="653" alt="image" src="https://user-images.githubusercontent.com/45538723/145312890-dcf48763-c78c-4ca8-8cbb-62747cacc9d2.png">

Reply:
Irrelevant for MVP I think: focus on good design and working simulation rather than safety/legality of agent actions

Reply:
It will make it harder to review code since we have to check people don't override by mistake. Will leave it away for now though since we need to get a working MVP out

Reply:
Added TODO